src/puppetlabs/puppetdb/core.clj

```diff
@@ -22,10 +24,11 @@
   "Return a set of namespaces underneath the .cli parent"
   []
   {:post [(set? %)]}
-  (set (for [namespace (ns/find-namespaces-on-classpath)
-             :let [ns-str (name namespace)]
-             :when (.startsWith ns-str ns-prefix)]
-         namespace)))
+  (set (filter
+         #(str/starts-with?
+            (name %)
+            ns-prefix)
+         (find-namespaces (cp/classpath)))))
```

Comment:
Might suggest condensing this a bit, say something like
```clojure
(set (filter #(str/starts-with? (name %) ns-prefix)
             (find-namespaces (cp/classpath))))
```
or
```clojure
(->> (find-namespaces (cp/classpath))     
     (filter #(str/starts-with? (name %) ns-prefix))
     set)
```

Reply:
I like how the `->>` version reads, I'll change it to that.